spec.md

```diff
@@ -1616,23 +1617,34 @@ A register is initialized with an indeterminate value (see [@sec:indeterminate-v
+I.e., the operations within the region execute depending on the value of the condition and the kind of operation.
 
--   Connections (see [@sec:connections]) only drive their sink values when the product (logical and) of all conditions that are parents of the connect, but not parents of the sink declaration evaluates to true.
--   Commands (see [@sec:commands]) only evaluate when the product (logical and) of all parent conditions evaluates to true.
--   The evaluation of all other operations in conditional regions is unaffected by the condition.
-    E.g., register, wire, and node declarations as well as module instantiation statements and primitive operation expressions may occur in conditional regions but never evaluate conditionally.
+## Conditional Execution
+
+The *conditional execution* of operations in the region of a conditional statement is operation dependent:
+
+-   Connections (see [@sec:connections]) only execute when the product (logical and) of all conditions that are parents of the connect, but not parents of the sink declaration evaluates to true.
+-   Commands (see [@sec:commands]) only execute when the product (logical and) of all parent conditions evaluates to true.
+-   The execution of all other operations in conditional regions is unaffected by the condition.
+    E.g., register, wire, and node declarations as well as module instantiation statements and primitive operation expressions may occur in conditional regions but never execute conditionally.
 
 > These semantics may cause different behavior for trivial inlining (substitution of an instantiation's module body in place of the instantiation).
-> A register in a conditional region and an instantiation in a conditional region where the instantiated module contains a register will evaluate the same after a trivial inlining.
-> A command in a conditional region and a module instantiation in a conditional region where the instantiated module contains a command will not evaluate the same after a trivial inlining.
+> A register in a conditional region and an instantiation in a conditional region where the instantiated module contains a register will execute the same after a trivial inlining.
+> A command in a conditional region and a module instantiation in a conditional region where the instantiated module contains a command will not execute the same after a trivial inlining.
 > In this latter case, the command is not conditional before trivial inlining and conditional after trivial inlining.
 
-## When Statements
+## Conditional Statements
+
+FIRRTL provides several types of conditional statements.
+
+### When Statements
 
-Connections within a when statement that connect to previously declared components hold only when the given condition is high.
-The condition must have a 1-bit unsigned integer type.
+When statements define a condition and two conditional regions: an "if" region and an "else" region.
```

Comment:
The term "if" region seems a bit weird, since we call them `when` statements.

Also, you're calling them "regions", but wouldn't "branches" be more conventional?

Reply:
The FIRRTL Dialect term here is "then" and "else". How does that sound?

Reply:
I'm leaning on "region" here as it's the more general term. E.g., match statements have a number of regions that are all conditioned on a condition. Layer blocks also have regions. Modules really have a single region, too.

There's prior art here where MLIR's SCF (structured control flow) actually calls them "regions" for it's `if` operation: https://mlir.llvm.org/docs/Dialects/SCFDialect/#scfif-scfifop

Reply:
Changed to "then" and "else" region language here: https://github.com/chipsalliance/firrtl-spec/pull/215/commits/f0f62dfa1cd9a6fb6be83dab9c202a8dc25f3021